build_scripts/api/dreambooth/utils/image_utils.py

```
@@ -0,0 +1,538 @@
+from __future__ import annotations
+
+import hashlib
+import math
+import os
+import random
+import re
+import sys
+from io import StringIO
+
+from diffusers.schedulers import KarrasDiffusionSchedulers
+
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+from PIL import features, PngImagePlugin, Image, ExifTags
+
+import os
+from typing import List, Tuple, Dict, Union
+
+import numpy as np
+import torch
+
+from dreambooth.dataclasses.db_concept import Concept
+from dreambooth.dataclasses.prompt_data import PromptData
+from helpers.mytqdm import mytqdm
+from dreambooth import shared
+from dreambooth.shared import status
+
+
+def get_dim(filename, max_res):
+    with Image.open(filename) as im:
+        im = rotate_image_straight(im)
+        width, height = im.size
+        if width > max_res or height > max_res:
+            aspect_ratio = width / height
+            if width > height:
+                width = max_res
+                height = int(max_res / aspect_ratio)
+            else:
+                height = max_res
+                width = int(max_res * aspect_ratio)
+        return width, height
+
+
+def rotate_image_straight(image: Image) -> Image:
+    try:
+        exif: Image.Exif = image.getexif()
+        if exif:
+            orientation_tag = {v: k for k, v in ExifTags.TAGS.items()}['Orientation']
+            orientation = exif.get(orientation_tag)
+            degree = {
+                3: 180,
+                6: 270,
+                8: 90,
+            }.get(orientation)
+            if degree:
+                image = image.rotate(degree, expand=True)
+        # else:
+        #     print(f"No exif data for {image.filename}. Using default orientation.")
+    except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```